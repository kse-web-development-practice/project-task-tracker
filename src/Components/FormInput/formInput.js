import React from 'react'
import PropTypes from 'prop-types'
import styles from './formInput.module.css'

export const FormInput = ({ labelText, type, value, onChange }) => {
  return (
    <div className={styles.formDiv}>
      <label className={styles.formLabel}>{labelText}</label>
      <input className={styles.formInput} type={type} value={value} onChange={onChange} />
    </div>
  )
}

FormInput.propTypes = {
  labelText: PropTypes.string,
  type: PropTypes.string.isRequired,
  value: PropTypes.string,
  onChange: PropTypes.func
}
