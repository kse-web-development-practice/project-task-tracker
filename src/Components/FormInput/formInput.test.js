import React from 'react'
import { render } from '@testing-library/react'
import { FormInput } from './formInput'

describe('FormInput Component', () => {
  test('default', () => {
    const { asFragment } = render(
      <FormInput labelText="Username" type="text" value="" onChange={() => {}} />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('password', () => {
    const { asFragment } = render(
      <FormInput labelText="Password" type="password" value="" onChange={() => {}} />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('date', () => {
    const { asFragment } = render(
      <FormInput labelText="Date" type="date" value="" onChange={() => {}} />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
