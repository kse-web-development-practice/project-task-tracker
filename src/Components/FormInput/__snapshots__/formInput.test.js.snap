// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput Component date 1`] = `
<DocumentFragment>
  <label
    class="formLabel"
  >
    Date
    <input
      class="formInput"
      type="date"
      value=""
    />
  </label>
</DocumentFragment>
`;

exports[`FormInput Component default 1`] = `
<DocumentFragment>
  <label
    class="formLabel"
  >
    Username
    <input
      class="formInput"
      type="text"
      value=""
    />
  </label>
</DocumentFragment>
`;

exports[`FormInput Component password 1`] = `
<DocumentFragment>
  <label
    class="formLabel"
  >
    Password
    <input
      class="formInput"
      type="password"
      value=""
    />
  </label>
</DocumentFragment>
`;
